File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_diff=0
        lowest=prices[0]

        for i in range(len(prices)-1):
            lowest=min(lowest, prices[i])
            if prices[i+1] > prices[i]:
                max_diff = max(prices[i+1]-lowest, prices[i+1]-prices[i], max_diff)

        return max_diff
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let diff = 0
    let lowest = prices[0]

    for (let i = 0; i < prices.length - 1; i++) {
        lowest = Math.min(lowest, prices[i + 1])
        if (prices[i + 1] - prices[i] > 0) {
            diff = Math.max(diff, prices[i + 1] - prices[i], prices[i + 1] - lowest)
        };
    ;}
    
    return diff;
};